tests_integration/__tests__/doc-trim.js

```diff
@@ -0,0 +1,94 @@
+"use strict";
+
+const prettier = require("prettier/local");
+const docPrinter = prettier.doc.printer;
+const docBuilders = prettier.doc.builders;
+
+const printDocToString = docPrinter.printDocToString;
+const concat = docBuilders.concat;
+const line = docBuilders.line;
+const trim = docBuilders.trim;
+const group = docBuilders.group;
+const indent = docBuilders.indent;
+const markAsRoot = docBuilders.markAsRoot;
+
+// These tests don't use `runPrettier` because `trim` is not used by any
+// bundled parser (only third-party plugins).
+
+describe("trim", () => {
+  test.each([
+    [
+      // trims the current line
+      group(concat(["hello    ", trim])),
+      "hello"
+    ],
+    [
+      // trims existing indentation
+      group(
+        concat([
+          "function()",
+          line,
+          "{",
+          indent(
+            concat([
+              line,
+              group(concat([trim, "#if DEBUG"])),
+              line,
+              "alert(42);",
+              line,
+              group(concat([trim, "#endif"]))
+            ])
+          ),
+          line,
+          "}"
+        ])
+      ),
+      `function()
+{
+#if DEBUG
+  alert(42);
+#endif
+}`
+    ],
+    [
+      // trims up to the the first column, even if there's indented root.
+      group(
+        concat([
+          "{",
+          indent(
+            concat([
+              markAsRoot(line),

```

Comment:
It seems this `markAsRoot` has no effect, it won't affect anything.

I'd expect something like:

```js
[
  // hardline will insert a newline with current indentation
  concat([indent(markAsRoot(indent(hardline))), "123"]),
  "\n    123"
],
[
  // literalline will insert a newline with root indentation
  concat([indent(markAsRoot(indent(literalline))), "123"]),
  "\n  123"
],
[
  // trims up to the the first column, even if there's indented root.
  concat([indent(markAsRoot(indent(literalline))), trim, "123"]),
  "\n123"
],
```

Reply:
Ok, will do. Thanks for your time :)

Reply:
These tests won’t pass as is. Did you mean that you expected them to pass and that `markAsRoot` seems broken?

Anyway, even if they passed, they should probably be part of another test file, say `doc-mark-as-root.js`. But in that case, I’m not sure what I should do. Maybe tests for `markAsRoot` should be added in a separate PR, where we clarify what is expected?

Reply:
I pushed the tests above along with my changes to see if it breaks anything on a different environment. I've pushed these changes from a Windows machine, so that could be a line ending problem.